lib/src/local_working_copy.rs

```diff
@@ -104,19 +107,117 @@ use crate::working_copy::WorkingCopy;
 use crate::working_copy::WorkingCopyFactory;
 use crate::working_copy::WorkingCopyStateError;
 
+/// The executable bit for a filetype, potentially ignored.
+///
+/// On Windows there is no executable bit, so this will always be Ignore. On
+/// Unix it will usually be Exec(true|false), but may be ignored by a config
+/// value or by a check we run when we load the config if not specified.
+#[derive(Debug, Clone, Copy)]
+pub enum ExecFlag {
+    Exec(bool),
+    Ignore,
+}
```

Comment:
Is it possible to always use a bool instead of this type and only change how we used that bool based on the `IgnoreExec`?

Reply:
We could, the main difference is just whether the tree state stores that a file is not executable at a given moment. I assume this would affect users who modify the configuration between snapshots, but I don't really know what that would imply.

I can convert it to just a bool if you'd like.

Reply:
Alternatively, it might be better to use a bool-equivalent enum, e.g. `enum { Exec, NotExec }`.

Reply:
> We could, the main difference is just whether the tree state stores that a file is not executable at a given moment. I assume this would affect users who modify the configuration between snapshots, but I don't really know what that would imply.

Good point. I think that means it's better to do our best to keep the stored flag up to date even the configuration currently says to ignore the executable bit in the file system. Otherwise we might clear the flag in the file and not notice if the user then sets the config to respect the config and also has cleared the executable bit on some file in the file system.

> Alternatively, it might be better to use a bool-equivalent enum, e.g. `enum { Exec, NotExec }`.

I would go with just a `bool` until we find that it's hard to read or likely to cause bugs for some reason (e.g. because we confuse that boolean for some other boolean value).